specification/resources/resource-manager/Microsoft.Management/preview/2018-01-01-preview/management.json

```diff
@@ -0,0 +1,1266 @@
+{
+  "swagger": "2.0",
+  "host": "management.azure.com",
+  "info": {
+    "version": "2018-01-01-preview",
+    "title": "Management Groups API",
+    "description": "The Azure Management Groups API enables consolidation of multiple \nsubscriptions/resources into an organizational hierarchy and centrally \nmanage access control, policies, alerting and reporting for those resources.\n"
+  },
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "tags": [
+    {
+      "name": "ManagementGroups",
+      "description": "A Management Group is a customer defined scope (grouping mechanism) that \nprovides access control (authorization), policy management and reporting. \nManagement Groups are organized in a strictly tree-based hierarchy.\n"
+    },
+    {
+      "name": "Operations",
+      "description": "Management operations supported by the Microsoft.Management resource provider.\n"
+    },
+    {
+      "name": "Entities",
+      "description": "A list of entities that belong to the Management Groups.\n"
+    }
+  ],
+  "paths": {
+    "/providers/Microsoft.Management/managementGroups": {
+      "get": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "ManagementGroups_List",
```

Comment:
>ManagementGroups_List [](start = 24, length = 21)

Odata extension could be useful here

Reply:
We only do a subset of Odata operations and the generators complain due to a lack of some parameters. `e.g. "Method with x-ms-odata needs to have "$filter" parameter.`